### ISSUE TITLE
server profile connection settings

### ISSUE THREAD

**Issue by rismoney** [HewlettPackard/terraform-provider-oneview#238]
Is there a way to specify connection settings for server profiles?  Can this be added as well? Example

"name" : "connection1",
"functionType" : "Ethernet",
"portId" : "Auto",
"requestedMbps" : 2500,
"boot" : {
"priority" : "Secondary",
"ethernetBootType" : "PXE"

**Comment by rismoney** [comment]
@nabhajit-ray is this functionality able to be added as well?

**Comment by rismoney** [comment]
this looks like it is in there, but not documented.

**Comment by rismoney** [comment]
data "oneview_ethernet_network" "deployment" {
  name = "ETH_VLAN_1"
}

```
resource "oneview_server_profile" "default" {
...
  network = [{
    id = 1
    name = "Network_1"
    function_type = "Ethernet"
    network_uri = "${data.oneview_ethernet_network.deployment.uri}"
    port_id = "Mezz 3:1-a"
    boot = {
      priority = "Primary"
      ethernet_boot_type = "PXE"
    }
  }]
...
```

This apparently does not error out, but the network connection is not being created.
fyi - @AsisBagga @nabhajit-ray

**Comment by rismoney** [comment]
Really need to get this capability working with network assignments or we will need to abandon.

**Comment by AsisBagga** [comment]
@rismoney Hi Rich, I will look into this today, and will update you. 
However I can suggest a work around, you can try creating server profile template and assign it to a Server Profile, you may refer this [example](https://github.com/HewlettPackard/terraform-provider-oneview/blob/master/examples/demos/infrastructure_provisioning_with_i3s.tf)

**Comment by rismoney** [comment]
Thanks for your immediate attention to this!  I cannot use profile templates for networking configuration.  I have too many disparate network configurations in the profiles, so we're only keeping the base onfigurations in templates (local storage and bios effectively).  Otherwise I effectively lose any benefit of template reuse.

**Comment by AsisBagga** [comment]
@rismoney ok, i have fixed the connection issue and incorporated the changes to #269  PR.
However, just as a side note, the master branch for this SDK will be upgraded to Terraform 0.12 soon, hence if you want to continue using terraform 0.11 version, then you may refer branch [Terraform_0.11](https://github.com/HewlettPackard/terraform-provider-oneview/tree/Terraform-0.11) (the following changes will be updated to Terraform 0.11 branch soon, therefor for now, you may refer PR)
And Syntax for passing the connection attributes while creating SP and SPT has changed, please have a look into that as well. 
[Example File](https://github.com/HewlettPackard/terraform-provider-oneview/blob/a37306191dc40b7d50d09d0c5e7e6f8fd567090a/examples/server_profile.tf)

**Comment by rismoney** [comment]
I can confirm #269 now works with network connection!  Thank you!  This is great news and I appreciate all the hard work you guys have put forth to get this functional.

How soon for .12? :)   I am definitely going to want to run with 12+ since I think I am going to need the dynamic looping capability to pass connection_settings arrays into a module.

**Comment by AsisBagga** [comment]
@rismoney This is fix is available for Terraform v0.12 now, Ill close this issue, if you have any query please feel free to open a new issue.  Thank you.
